styx-common/src/main/java/com/spotify/styx/util/ReplayEvents.java

```diff
@@ -56,7 +57,19 @@ private ReplayEvents() {
 
     return instances.entrySet().parallelStream().map(entry -> {
       final WorkflowInstance workflowInstance = entry.getKey();
-      final long lastConsumedEvent = entry.getValue();
+      final PersistentWorkflowInstanceState persistentState = entry.getValue();
+      final long lastConsumedEvent = persistentState.counter();
+
+      // TODO: always use the persistent state from datastore when it has been migrated
+      if (persistentState.state() != null) {
+        if (printLogs) {
+          LOG.debug("Using persistent state instead of replaying events");
```

Comment:
This logging is not very useful in this form, let's add the processed WorkflowInstance as well?